### PR TITLE
Commit API docs and generate them using a separate task.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
# Description

Currently, clp-ffi-py's API docs are built whenever the docs site is built. This has two issues:

* We can't apply any manual customizations to the API docs, if we wanted to.
* The edit links in the docs site won't work for the API docs since they aren't committed to the repo.

This PR commits the API docs to the repo and builds them using a separate task that devs should run manually if they ever add/remove a package/module.

This PR depends on #46 being merged first.

# Validation performed

* Validated `task docs:serve` continues to work correctly.
* Validated `task docs:api-docs` generates the API docs but doesn't change the committed docs (right now).

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -21,14 +21,29 @@ documentation website for this package:
...
   task docs:clean
   ```
 
+* Generate API docs:
+
+  ```shell
+  task docs:api-docs
+  ```
+  
+  * This will regenerate the API docs in `src/api`. You should run this whenever you change what
```
Comment (haiqi96) [inline docs/README.md]: do we need the "what" in the sentence?

File: docs/src/api/clp_ffi_py.ir.native.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+clp\_ffi\_py.ir.native module
```
Comment (haiqi96) [inline docs/src/api/clp_ffi_py.ir.native.rst:1]: I have no idea what are those so I assume they are auto generated

**Review by haiqi96 — COMMENTED** [review]
Only a small comment on the grammer

File: docs/src/api/clp_ffi_py.ir.native.rst (hunk 2 above)
Comment (kirkrodrigues) [reply docs/src/api/clp_ffi_py.ir.native.rst:1]: Yessir.

File: docs/README.md (hunk 1 above)
Comment (kirkrodrigues) [reply docs/README.md]: Updated the wording.

**Review by haiqi96 — APPROVED** [review]

**Pull request merged**